dev/io.openliberty.microprofile.openapi.2.0.internal/src/com/ibm/ws/microprofile/openapi/servlet/filter/HtmlResponseWrapper.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+
+package com.ibm.ws.microprofile.openapi.servlet.filter;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+/**
+ * Response Wrapper to capture the Servlet Response to allow for rewriting of the contents
+ */
+public class HtmlResponseWrapper extends HttpServletResponseWrapper {
+
+    private final ByteArrayOutputStream baos;
+    private ServletOutputStream outputStream = null;
+    private PrintWriter writer = null;
+
+    public HtmlResponseWrapper(HttpServletResponse response) {
+        super(response);
+        baos = new ByteArrayOutputStream(response.getBufferSize());
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() {
+        if (outputStream == null) {
+            outputStream = new ServletOutputStream() {
+                @Override
+                public void write(int b) throws IOException {
+                    baos.write(b);
+                }
+                @Override
+                public void flush() throws IOException {
+                    baos.flush();
+                }
+                @Override
+                public void close() throws IOException {
+                    baos.close();
+                }
+                @Override
+                public boolean isReady() {
+                    return false;
+                }
+                @Override
+                public void setWriteListener(WriteListener var1) {
+                }
+            };
+        }
+
+        return outputStream;
+    }
+
+    @Override
+    public PrintWriter getWriter() throws IOException {
+        if (writer == null) {
+            writer = new PrintWriter(new OutputStreamWriter(baos, getCharacterEncoding()));
+        }
+
+        return writer;
+    }
```

Comment:
Although the use case for this filter is limited, it would be good to throw `IllegalStateException` if both methods are called.